docs/proposals/nvidia-gpu-enhancement.md

```diff
@@ -0,0 +1,64 @@
+# Add NVIDIA GPU discovery and assignment capabilities to Kubelet.
+
+## Background:
+
+So far we only support 1 GPU(```/dev/nvidia0```) and the GPU device need to be configured as a Kubelet's parameter manually. But in production environment, there could have multiple GPUs(```/dev/nvidia0, /dev/nvidia1, /dev/nvidia2 ...```) installed on 1 Minion, also 1 micro service could needs more than 1 GPU, and there could have thousands of Minions which means if Kubelet could discover GPU by itself would be great.
+
+
+## Goal:
+
+ - Have the capabilities of GPU discovery to let Kubelet could aware more than 1 NVIDIA GPU and make the deployment easier.
+ - Assign more than 1 GPU to 1 container to make the micro service with multiple GPUs requirement run on Kubernetes possible.
+ - Add a general GPU interface for all the GPU vendors for future use.
+
+
+## Implementation:
+
+### Limitation:
+
+In this implementation, we still only schedule dedicated NVIDIA GPU and NVIDIA GPU supported only, and will not be NVIDIA GPU details(familiies, cores, memory) sensitive.
```

Comment:
suggest:

"In this implementation, an application may only request a number of GPUs. Additional resource dimensions, such as GPU model, memory available, or number of cores, will not be able to be specified by an application.

The implementation will also only support NVIDIA GPUs."